CIP-0015/CIP-0015.md

```diff
@@ -33,57 +33,112 @@ A voting key is simply an ED25519 key. How this key is created is up to the wall
+
 Recall: Cardano uses the UTXO model so to completely associate a wallet's balance with a voting key (i.e. including enterprise addresses), we would need to associate every payment key to a voting key individually. Although there are attempts at this (see [CIP8](../CIP-0008/CIP-0008.md)), the resulting data structure is a little excessive for on-chain metadata (which we want to keep small)
 
 Given the above, we choose to only associate staking keys with voting keys. Since most Cardano wallets only use base addresses for Shelley wallet types, in most cases this should perfectly match the user's wallet.
 
 ### Registration metadata format
 
-A Catalyst registration transaction are a regular Cardano transaction with a specific transaction metadata associated with it
+A Catalyst registration transaction is a regular Cardano transaction with a specific transaction metadata associated with it.
 
-Notably, there should be three entries inside the metadata map:
+Notably, there should be four entries inside the metadata map:
 
-Voting key registration
+Voting registration example:
 ```
 61284: {
   // voting_key - CBOR byte array
   1: "0xa6a3c0447aeb9cc54cf6422ba32b294e5e1c3ef6d782f2acff4a70694c4d1663",
   // stake_pub - CBOR byte array
   2: "0xad4b948699193634a39dd56f779a2951a24779ad52aa7916f6912b8ec4702cee",
-  // address - CBOR byte array
-  3: "0x00588e8e1d18cba576a4d35758069fe94e53f638b6faf7c07b8abd2bc5c5cdee47b60edc7772855324c85033c638364214cbfc6627889f81c4"
+  // reward_address - CBOR byte array
+  3: "0x00588e8e1d18cba576a4d35758069fe94e53f638b6faf7c07b8abd2bc5c5cdee47b60edc7772855324c85033c638364214cbfc6627889f81c4",
+  // slot_number
+  4: 5479467
 }
 ```
 
-Signing the voting public key with the staking private key
+The entries under keys 1, 2, 3, and 4 represent the Catalyst voting key,
+the staking key on the Cardano network, the address to receive rewards,
+and the current slot number, respectively. A registration with these metadata
+will be considered valid if the following conditions hold:
+
+- The slot number of the block including this transaction is equal to
+  the slot number value under key 4 or its increment by one.
+- The reward address is a Shelley address discriminated for the same network
+  this transaction is submitted to.
+
+To produce the signature field, the CBOR representation of a map containing
+a single entry with key 61284 and the registration metadata map in the
+format above is formed, designated here as `sign_data`.
+This data is signed with the staking key using one of the two methods,
+discriminated by the key number in the signature metadata map:
+
+1: `sign_data` is signed as is using Ed25519.
+
+2: The blake2b-256 hash of `sign_data` is signed using Ed25519.
+
+Signature example:
 ```
 61285: {
   // signature - ED25119 signature CBOR byte array
   1: "0x8b508822ac89bacb1f9c3a3ef0dc62fd72a0bd3849e2381b17272b68a8f52ea8240dcc855f2264db29a8512bfcd522ab69b982cb011e5f43d0154e72f505f007"
```

Comment:
shouldn't the key be updated to "2"?

Reply:
The example is still valid, applications should be free to use scheme 1 if they prefer.

Reply:
I see, then as pointed out in my above comment, it would be worth clarifying what's the expected behavior if somebody specifies both keys - I'd suggest discarding such registrations as invalid

Reply:
no, it should use 1. 2 will be ignored by the snapshot tools. We will not support both signature options.